velox/connectors/hive/HiveConnector.cpp

```diff
@@ -166,7 +193,12 @@ HiveDataSource::HiveDataSource(
   readerOutputType_ = ROW(std::move(columnNames), std::move(outputTypes));
   scanSpec_ =
       makeScanSpec(hiveTableHandle->subfieldFilters(), readerOutputType_);
-
+  if (readAsLowerCase) {
+    for (auto& child : scanSpec_->children()) {
```

Comment:
You should check the field names before we convert it to `ScanSpec`, i.e. check `hiveTableHandle->subfieldFilters()`